Source/Core/VideoCommon/CPMemory.h

```diff
@@ -18,176 +20,239 @@ enum
 };
 
 // Vertex components
-enum
+enum class VertexComponentFormat
+{
+  NotPresent = 0,
+  Direct = 1,
+  Index8 = 2,
+  Index16 = 3,
+};
+template <>
+struct fmt::formatter<VertexComponentFormat> : EnumFormatter<VertexComponentFormat::Index16>
 {
-  NOT_PRESENT = 0,
-  DIRECT = 1,
-  INDEX8 = 2,
-  INDEX16 = 3,
+  formatter() : EnumFormatter({"Not present", "Direct", "8-bit index", "16-bit index"}) {}
+};
 
-  MASK_INDEXED = 2,
+enum class ComponentFormat
+{
+  UByte = 0,   // 2 Cmp - Invalid for normals
+  Byte = 1,    // 3 Cmp
+  UShort = 2,  // Invalid for normals
+  Short = 3,
+  Float = 4,
+};
```

Comment:
Anyone know what the `2 Cmp` and `3 Cmp` comments might be referring to?  They've existed since [the initial megacommit](https://github.com/dolphin-emu/dolphin/blame/775dc8a9c0c83f7a2f776e97f5b90385c759535a/Source/Plugins/Plugin_VideoDX9/Src/CPStructs.h#L68-L69) (along with the `NA` comment on `FORMAT_16B_565` which I've removed).  The best guess I have is that "cmp" refers to "component" and the labels should instead apply to `CoordComponentCount`'s `XY` and `XYZ`, but that seems unlikely.  [Yagcd](http://hitmen.c02.at/files/yagcd/yagcd/chap5.html#sec5.11.2) doesn't mention anything like this either (ctrl+f Vertex Attribute Data Formats)

Reply:
My take is that these were comments made in observation during reverse engineering.  I don't think they have any direct relevance to the datatype format.  But @hrydgard is still around.  Maybe he'll remember :D

Reply:
I honestly don't remember. You're probably best off just deleting those comments, much wasn't understood properly back then...